docs/src/main/sphinx/routines/introduction.md

```diff
@@ -0,0 +1,159 @@
+# Introduction to SQL routines
+
+A SQL routine is a custom, user-defined function authored by a user of Trino in
+a client and written in SQL. You can declare the routine body within a
+[](/routines/function) block as inline routines or catalog routines.
+
+(routine-inline)=
+## Inline routines
+
+An inline routine declares and uses the routine within a query processing
+context. The routine is declared in a `WITH` block before the query:
+
+```sql
+WITH
+  FUNCTION abc(x integer)
+    RETURNS integer
+    RETURN x * 2
+SELECT abc(21);
+```
+
+Inline routine names must follow SQL identifier naming conventions, and cannot
+contain `.` characters.
+
+The routine declaration is only valid within the context of the query. A
+separate later invocation of the routine is not possible, without declaring the
+routine again with the query.
+
+Multiple inline routine declarations are comma-separated, and can include
+routines calling each other, as long as a called routine is declared before
+the first invocation.
+
+```sql
+WITH
```

Comment:
@electrum and @martint .. do we have to update the SELECT documentation for this. I think yes .. please confirm

Reply:
Confirmed after chat with Martin